haystack/components/builders/chat_prompt_builder.py

```diff
@@ -171,21 +171,30 @@ def __init__(
 
         extracted_variables = []
         if template and not variables:
+
+            def _extract_from_text(
```

Comment:
Could we make this a class function instead of inline. Please also keep it as private so fine to leave the name as `_extract_from_text`

Reply:
Hey @srini047 it looks like this function wasn't made into a class function with signature
```python
def _extract_from_text(
    self, text: Optional[str], role: Optional[str] = None, is_filter_allowed: bool = False
) -> list[str]:
```
yet